internal/providers/pluginfw/tfschema/customizable_schema.go

```diff
@@ -11,39 +11,45 @@ import (
 
 // CustomizableSchema is a wrapper struct on top of AttributeBuilder that can be used to navigate through nested schema add customizations.
 type CustomizableSchema struct {
-	attr AttributeBuilder
+	attr BaseSchemaBuilder
 }
 
 // ConstructCustomizableSchema constructs a CustomizableSchema given a map from string to AttributeBuilder.
-func ConstructCustomizableSchema(attributes map[string]AttributeBuilder) *CustomizableSchema {
-	attr := AttributeBuilder(SingleNestedAttributeBuilder{Attributes: attributes})
+func ConstructCustomizableSchema(nestedObject NestedBlockObject) *CustomizableSchema {
```

Comment:
Make sure to update the doccomments for all of these methods as they refer to the old types, and also for CustomizableSchema.